File: CS-Examples/04_RowsColumns/CopyRows.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace CopyRows
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing workbook with a pivot table from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Copying.xls");

            // Get the second worksheet in the workbook
            Worksheet sheet1 = workbook.Worksheets[1];

            // Get the first worksheet in the workbook
            Worksheet sheet2 = workbook.Worksheets[0];

            // Copy the first row (row index 0) to the third row (row index 2) in the same sheet
            sheet1.Copy(sheet1.Rows[0], sheet1.Rows[2], true, true, true);

            // Copy the first row (row index 0) to the second row (row index 1) in a different sheet
            sheet1.Copy(sheet1.Rows[0], sheet2.Rows[1], true, true, true);

            // Specify the output file name for the result
            string result = "CopyRows_result.xlsx";

            // Save the modified workbook to the specified file using Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/04_RowsColumns/CopyRows.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace CopyRows
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Instantiate a new Workbook object to create a workbook.
            Dim workbook As New Workbook()

            'Load the Excel file, which includes a pivot table, from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Copying.xls")
            'Get the second worksheet from the workbook.
            Dim sheet1 As Worksheet = workbook.Worksheets(1)
            'Get the first worksheet from the workbook.
            Dim sheet2 As Worksheet = workbook.Worksheets(0)

            'Copy the content of the first row to the third row in the same sheet.
            sheet1.Copy(sheet1.Rows(0), sheet1.Rows(2), True, True, True)


            'Copy the content of the first row in sheet1 to the second row in sheet2.
            sheet1.Copy(sheet1.Rows(0), sheet2.Rows(1), True, True, True)

            'Specify the name for the resulting file.
            Dim result As String = "CopyRows_result.xlsx"

            'Save the workbook to a file with the specified name and Excel version (in this case, Excel 2010).
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
